src/bson.rs

```diff
@@ -402,16 +493,22 @@ impl Bson {
 
     /// Converts to extended format.
     /// This function mainly used for [extended JSON format](https://docs.mongodb.com/manual/reference/mongodb-extended-json/).
-    #[doc(hidden)]
-    pub fn to_extended_document(&self) -> Document {
+    pub(crate) fn to_extended_document(&self) -> Document {
```

Comment:
This function is a bit weird; it takes a BSON value and converts it into a document that represents the extended JSON, but still is BSON rather than JSON. I originally played around with getting rid of it, but it's used in deep parts of the serde implementation, so to avoid scope creep, I kept it around but converted it to mirror the new extended JSON implementation. I also figured that instead of just hiding it from the docs, it was better to just not export it. On the off-chance that we decide it needs to be part of the API again, changing it back to `pub` isn't a breaking change, so we could do that later.

Reply:
Filed RUST-426 and added the TODO.